### PR TITLE
Hw3 t03

### PR DESCRIPTION
**Pull request by mirmozavr**
Filter function

### CODE REVIEW

File: hw3/tests/test_hw3_t03.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    ]
+    assert (
+        make_filter(
+            {
+                "name": "Bill",  # noqa: F601
+                "name": "polly",  # noqa: F601
+            }
+        ).apply(sample_data)
+        == []
+    )
```
Comment (mirmozavr) [inline hw3/tests/test_hw3_t03.py]: This one.

File: hw3/hw3_t03.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+class Filter:
+    """
+    Helper filter class.
+
+    Accepts a list of single-argument
+    functions that return True if object in list conforms to some criteria
+    """
+
+    def __init__(self, *args: Callable):
+        self.functions = list(args)
```
Comment (Cjkjvfnby) [inline hw3/hw3_t03.py]: Why list?

File: hw3/tests/test_hw3_t03.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
+from hw3.hw3_t03 import make_filter
+
+
+def test_polly():
```
Comment (Cjkjvfnby) [inline hw3/tests/test_hw3_t03.py]: Who is Polly?   

This is no a filter for characters, it is a general-purpose filter. Characters are used as an example.

You have key, value, and item, so please build your positive and negative test around these terms.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
Please make test names according to my comments. 

Please remove unused items from tests fixtures if it is not a subject of testing.  Keep test items as small as possible.

File: hw3/hw3_t03.py (hunk 2 above)
Comment (mirmozavr) [reply hw3/hw3_t03.py]: List is ready for addition. But in this case, it should stay a tuple.

File: hw3/tests/test_hw3_t03.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
+from hw3.hw3_t03 import make_filter
+
+
+def test_positive_for_1_subject():
```
Comment (Cjkjvfnby) [inline hw3/tests/test_hw3_t03.py:4]: It looks more like a COVID-19 test.

File: hw3/tests/test_hw3_t03.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+            "type": "person",
+        },
+        {"is_dead": True, "kind": "parrot", "type": "bird", "name": "polly"},
+    ]
+    assert make_filter({"name": "polly", "type": "bird"}).apply(sample_data) == [
+        {"is_dead": True, "kind": "parrot", "type": "bird", "name": "polly"}
+    ]
+
+
+def test_both_characters_by_diff_key_value_pairs():
```
Comment (Cjkjvfnby) [inline hw3/tests/test_hw3_t03.py:19]: Don't try to shorten words. Especially such commonly used in [IT](https://git-scm.com/docs/git-diff).

File: hw3/tests/test_hw3_t03.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            {
+                "occupation": "was here",
+                "type": "bird",
+            }
+        ).apply(sample_data)
+        == []
+    )
+
+
+def test_with_non_existent_value():
```
Comment (Cjkjvfnby) [inline hw3/tests/test_hw3_t03.py:40]: `non_existent` sounds way too philosophical. I'd prefer missing or absent.

**Review by Cjkjvfnby — APPROVED** [review]
I am not happy with the test names, but it is a task3.

**Pull request merged**